py/objtype.c

```diff
@@ -336,8 +336,10 @@ mp_obj_t mp_obj_instance_make_new(const mp_obj_type_t *self, size_t n_args, cons
             mp_obj_t *args2 = m_new(mp_obj_t, 1 + n_args + 2 * n_kw);
             args2[0] = MP_OBJ_FROM_PTR(self);
             memcpy(args2 + 1, args, n_args * sizeof(mp_obj_t));
-            // copy in kwargs
-            memcpy(args2 + 1 + n_args, kw_args->table, 2 * n_kw * sizeof(mp_obj_t));
+            if (n_kw) {
+                // copy in kwargs
+                memcpy(args2 + 1 + n_args, kw_args->table, 2 * n_kw * sizeof(mp_obj_t));
```

Comment:
What does this do?

Reply:
The diagnostic from clang scan-build was that in the memcpy statement, `kw_args->table` dereferenced null pointer `kw_args`.  If kw_args is null, then n_kw is 0.  By guarding it with the added `if` statement, the problem is resolved and the NULL pointer is not dereferenced at runtime.

Reply:
(and if n_kw is 0 then there's nothing to copy anyway)

Reply:
Ah, why not check kw_args for NULL then? `memcpy` should handle the n_kw = 0 ok.

Reply:
done